packages/ui/react-ui-table/src/util/table.ts

```diff
@@ -13,81 +13,108 @@ import {
   TypedObject,
   TypeEnum,
 } from '@dxos/echo-schema';
-import { log } from '@dxos/log';
 import { PublicKey } from '@dxos/react-client';
 import { create, makeRef, type Space } from '@dxos/react-client/echo';
-import { createFieldId, createView, ViewProjection } from '@dxos/schema';
+import { createFieldId, createView, getSchemaProperties, ViewProjection } from '@dxos/schema';
 
 import { type TableType } from '../types';
 
 type InitialiseTableProps = {
   space: Space;
   table: TableType;
   initialRow?: boolean;
+  initialSchema?: string;
 };
 
+// TODO(ZaymonFC): Clean up the branching in this file.
+
 // TODO(burdon): Pass in type.
 // TODO(burdon): User should determine typename.
 export const initializeTable = async ({
   space,
   table,
   initialRow = true,
+  initialSchema,
 }: InitialiseTableProps): Promise<EchoSchema> => {
-  log('initializeTable', { table });
+  if (initialSchema) {
+    const schema = await space.db.schemaRegistry.query({ typename: initialSchema }).firstOrUndefined();
 
-  const ContactSchema = TypedObject({
-    typename: `example.com/type/${PublicKey.random().truncate()}`,
-    version: '0.1.0',
-  })({
-    name: S.optional(S.String).annotations({
-      [AST.TitleAnnotationId]: 'Name',
-    }),
-    active: S.optional(S.Boolean),
-    email: S.optional(Format.Email),
-    salary: S.optional(Format.Currency()).annotations({
-      [AST.TitleAnnotationId]: 'Salary',
-    }), // TODO(burdon): Should default to prop name?
-  });
+    if (!schema) {
+      throw new Error(`Schema not found: ${initialSchema}`);
+    }
 
-  const [contactSchema] = await space.db.schemaRegistry.register([ContactSchema]);
+    // We need to get the schema properties here. For now, only simple types and refs, not compound types
+    // are going to be supported.
+    const fields = getSchemaProperties(schema.ast)
+      .filter((prop) => prop.type !== 'object' || prop.format === FormatEnum.Ref)
+      .map((prop) => prop.name);
 
-  table.view = makeRef(
-    createView({
-      name: 'Test',
-      typename: contactSchema.typename,
-      jsonSchema: contactSchema.jsonSchema,
-      fields: ['name', 'active', 'email', 'salary'],
-    }),
-  );
+    table.view = makeRef(
+      createView({
+        // TODO(ZaymonFC): Don't hardcode name?
+        name: 'View',
```

Comment:
could the name be omitted?

Reply:
It's required in the schema at the moment.